operator/controllers/lifecycle/keptnappversion/reconcile_workloadsstate.go

```diff
@@ -20,18 +19,30 @@ func (r *KeptnAppVersionReconciler) reconcileWorkloads(ctx context.Context, appV
 		LongName:  "Reconcile Workloads",
 	}
 
-	var newStatus []klcv1alpha3.WorkloadStatus
+	workloadInstanceList, err := r.getWorkloadInstanceList(ctx, appVersion.Namespace, appVersion.Spec.AppName)
+	if err != nil {
+		r.Log.Error(err, "Could not get workloads")
+		return apicommon.StateUnknown, r.handleUnaccessibleWorkloadInstanceList(ctx, appVersion)
+	}
+
+	newStatus := make([]klcv1alpha3.WorkloadStatus, 0, len(appVersion.Spec.Workloads))
 	for _, w := range appVersion.Spec.Workloads {
 		r.Log.Info("Reconciling workload " + w.Name)
-		workload, err := r.getWorkloadInstance(ctx, getWorkloadInstanceName(appVersion.Namespace, appVersion.Spec.AppName, w.Name, w.Version))
-		if err != nil && errors.IsNotFound(err) {
+		workloadStatus := apicommon.StatePending
+		found := false
+		instanceName := getWorkloadInstanceName(appVersion.Spec.AppName, w.Name, w.Version)
```

Comment:
can you please add some comments explaining why we do additional filtering on the returned list? As discussed yesterday, the reason for that is completely valid, but just to avoid any confusion for other maintainers :)